rclpy/rclpy/executors.py

```diff
@@ -286,14 +286,14 @@ def spin(self) -> None:
     def spin_until_future_complete(self, future: Future, timeout_sec: float = None) -> None:
         """Execute callbacks until a given future is done or a timeout occurs."""
         if timeout_sec is None or timeout_sec < 0:
-            while self._context.ok() and not future.done() and not self._is_shutdown:
+            while self._context.ok() and not future.done():
```

Comment:
I think this `and not self._is_shutdown()` check should be kept. IIUC if the executor is shut down but not the context then this would busy loop because spin_once would return immediately.